draft-ietf-oauth-attestation-based-client-auth.md

```diff
@@ -233,6 +234,57 @@ The following example is the decoded header and payload of a JWT meeting the pro
 }
 ~~~
 
+# Authorization Server-provided nonces
+
+This section specifies a mechanism using server provided opaque nonces to limit the lifetime of a Client Attestation PoP JWT, similar to the one defined in section 8 of {{RFC9449}}.
+Without employing such a mechanism, a malicious party controlling the client (potentially including the end-user) can create Client Attestation PoP JWT for use arbitrarily far in the future.
+In addition, a mismatch between the time reference of a non-malicious client (e.g. a mobile device) and the time reference of the authorization server, may result in Client Attestation PoP JWT that are never accepted by the authorization server.
+
+Including a nonce value contributed by the authorization server in the Client Attestation PoP JWT MAY be used by authorization servers to limit the lifetime of those proofs, both protecting against attacks and allowing for interoperability with clients with different time references.
+
+An authorization server MAY require the use of nonces.
+This requirement is communicated to the client by having the authorization server return an HTTP response with the `400` status and an `error` field with the value `"use_attestation_nonce"`, when the Client Attestation PoP JWT present in the HTTP request does not contain a valid nonce.
+This HTTP response MUST also contain the `Attestation-Nonce` header, with the value of the new server provided nonce.
+The client MUST use this new nonce value to create a new Client Attestation PoP JWT and resend the previous request.
```

Comment:
```suggestion
The Client MUST use this new nonce value to create a new Client Attestation PoP JWT and resend the previous request.
```
@pmhsfelix I'm wondering if using Client as defined term in OAuth 2.0. If yes this change is required within the entire text